library/java/net/openid/appauth/IdToken.java

```diff
@@ -160,15 +160,15 @@ private static JSONObject parseJwtSection(String section) throws JSONException {
         return new JSONObject(jsonString);
     }
 
-    static IdToken from(String token) throws JSONException, IdTokenException {
+    public static IdToken from(String token) throws JSONException, IdTokenException {
         String[] sections = token.split("\\.");
 
         if (sections.length <= 1) {
             throw new IdTokenException("ID token must have both header and claims section");
         }
 
         // We ignore header contents, but parse it to check that it is structurally valid JSON
```

Comment:
Please add an annotation to this comment that notes it is the original project comment, but that you've changed the code so that it is no longer valid. Then add a comment explaining that we are now getting the header contents and why (Azure sends the nonce through the header).

Reply:
changed the comment